### PR TITLE
escape columns in inheritance helper

### PR DESCRIPTION
**Pull request by AlternateIf**
in case of having a reserved keyword as field on an objectbrick (e.g "use") the inheritance helper fails to update a child that does not have a brick that is set on the parent. 

since this helper is called upon save of an object, this leads to the obect save throwing an exception in the backend. the dbal insert function states:
"Table expression and columns are not escaped and are not safe for user-input."%

the same might also apply for localized fields.

### CODE REVIEW

**Comment by mcop1** [comment]
Hello @AlternateIf ,
thanks for the pr. What do you think about using `quote` method of db class? And maybe `quoteDataIdentifiers` method of Helper class (https://github.com/pimcore/pimcore/blob/22b2ae4f38f759cd2f0612bdaf051124a233ebb9/lib/Db/Helper.php#L113) instead of the tmp array? 
Should have the same affect, right? And would help to reduce duplicated code.

**Comment by AlternateIf** [comment]
@mcop1 changed it to use the helper function

File: models/DataObject/Concrete/Dao/InheritanceHelper.php [hunk 1]
```diff
@@ -233,10 +234,11 @@ public function doUpdate(int $oo_id, bool $createMissingChildrenRows = false, ar
                 $missingIds = $this->db->fetchFirstColumn($query);
 
                 // create entries for children that don't have an entry yet
-                $originalEntry = $this->db->fetchAssociative('SELECT * FROM ' . $this->querytable . ' WHERE ' . $this->idField . ' = ?', [$oo_id]);
+                $originalEntry = Helper::quoteDataIdentifiers($this->db, $this->db->fetchAssociative('SELECT * FROM ' . $this->querytable . ' WHERE ' . $this->idField . ' = ?', [$oo_id]));
```
Comment (jdreesen) [inline models/DataObject/Concrete/Dao/InheritanceHelper.php:237]: I don't think this is how `Helper::quoteDataIdentifiers()` should be used :thinking:

**Comment by markus-moser** [comment]
LGTM - thanks for fixing it!

**Pull request merged**